pandas/tseries/tests/test_partial_string_indexes.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+
+import pandas.util.testing as tm
```

Comment:
Thanks. I'll move it to `test_timeseries.py`. Should I keep it in the separate class or unify with one of the existing? What do you mean under "new ways of testing"? Is it okay to add functions like `assert_exact` that I did?